src/Microsoft.AspNetCore.SignalR.Common/Internal/Protocol/JsonHubProtocol.cs

```diff
@@ -67,10 +66,12 @@ public void WriteMessage(HubMessage message, Stream output)
             TextMessageFormatter.WriteRecordSeparator(output);
         }
 
-        private HubMessage ParseMessage(Stream input, IInvocationBinder binder)
+        private HubMessage ParseMessage(TextReader input, IInvocationBinder binder)
         {
-            using (var reader = new JsonTextReader(new StreamReader(input)))
+            using (var reader = new JsonTextReader(input))
```

Comment:
Add an internal helper method for creating a `JsonTextReader` with shared array pool?